app/com/arpnetworking/metrics/portal/version_specifications/impl/NoPackagesVersionSpecificationRepository.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2016 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.version_specifications.impl;
+
+import com.arpnetworking.metrics.portal.version_specifications.VersionSpecificationRepository;
+import models.internal.QueryResult;
+import models.internal.VersionSet;
+import models.internal.VersionSetLookupResult;
+import models.internal.VersionSpecification;
+import models.internal.impl.DefaultQueryResult;
+import models.internal.impl.DefaultVersionSet;
+import models.internal.impl.DefaultVersionSetLookupResult;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * @author Matthew Hayter (mhayter at groupon dot com)
```

Comment:
Incomplete java doc. I'm ending the review here. Please polish this up and I'll have another look. If you already did a self pass consider having a peer review it as well.